### ISSUE TITLE
Display of partner preferences giving errors and not displaying

### ISSUE THREAD

**Issue by kinsey-nsfw** [Innoxia/liliths-throne-public#1284]
**Describe the bug**
When mousing over the starred icon to display a partner's preferences, nothing comes up, and the error log indicates a problem with getForeplayPreference/getMainSexPreference is at least part of it.

**Expected behavior**
As per earlier version behavior, showing partner preferences.

**To Reproduce**
Mouse over the starred icon (with hearts in the corners) to display a partner's preferences.

**Attach files**

See [this revision comment with an error.log link](https://github.com/Innoxia/liliths-throne-public/commit/4cdea7cab5faf3604c2926d87aa215c7f2d47e46#r36902996).

**Additional information:**
 - Game version: 0.3.6.1-dev
 - OS: OS X
 - Using .jar or .exe: .jar

**Comment by kinsey-nsfw** [comment]
Currently testing version with GameCharacter -> NPC rollback for interface.

**Comment by kinsey-nsfw** [comment]
Confirmed that rollback works, although I don't know if it's going to break anything else.

**Comment by AceXP** [comment]
I think the switch from NPC to GameCharacter was done because of the issue where Quick Sex uses all actions, including those that should be excluded because of the content preferences.
If that's the case, reverting back should not cause major issues. The content issue is not something that will break the game.

**Comment by kinsey-nsfw** [comment]
Sigh... no, doesn't work when doing sex with a slave in residence. Back to the drawing board...

**Comment by kinsey-nsfw** [comment]
And the breakage on reversion with a slave in residence locks up the game completely, so it's rather worse than the "tooltip" not working.

**Comment by kinsey-nsfw** [comment]
Aha! I think the problem is that there are still a lot of places where, for the two-character form of the getForeplayPreference/getMainSexPreference method, the first character is explicitly "cast" to "NPC" - including in StatusEffect.java. Trying de-reverting and then cleaning up the casts.

**Comment by kinsey-nsfw** [comment]
Yes, that works. Uploading my local dev version to my github fork; will try to turn into a consolidated pull request when I have a bit of time again...

**Comment by AceXP** [comment]
This seems to work as intended for some time now. Could you please check it and close this issue?
